src/librustc_typeck/collect.rs

```diff
@@ -690,22 +703,25 @@ fn super_predicates_of<'a, 'tcx>(
 
     let icx = ItemCtxt::new(tcx, trait_def_id);
 
-    // Convert the bounds that follow the colon, e.g. `Bar+Zed` in `trait Foo : Bar+Zed`.
+    // Convert the bounds that follow the colon, e.g. `Bar + Zed` in `trait Foo : Bar + Zed`.
     let self_param_ty = tcx.mk_self_type();
     let superbounds1 = compute_bounds(&icx, self_param_ty, bounds, SizedByDefault::No, item.span);
 
     let superbounds1 = superbounds1.predicates(tcx, self_param_ty);
 
     // Convert any explicit superbounds in the where clause,
     // e.g. `trait Foo where Self : Bar`:
```

Comment:
Nit: Can we extend the comment, like

In the case of trait aliases, though, we include **all** where clauses -- so e.g. `trait Foo = where u32: PartialEq<Self>` would include `u32: PartialEq<Self>` as one of its "superpredicates".

Reply:
I'm beginning to wonder a bit though whether we'll regret this — another option might be to leave this as a sort of "FIXME" to fix when we get implied bounds working with Chalk.

In particular, I am worried about "surprising" interactions with trait objects.

Reply:
Yeah, I worry about this too, but let's see how it goes... it's quite revertible for now.